### PR TITLE
bump tendermint 0.35

### PR DESCRIPTION
**Pull request by altergui**
this was the original upgrade in january v0.34.13 -> v0.35.1
 * https://github.com/vocdoni/vocdoni-node/pull/429
 * https://github.com/vocdoni/vocdoni-node/pull/455

a week later, @p4u rolled back to tendermint v0.34.13
 * https://github.com/vocdoni/vocdoni-node/pull/468
 
 now i'm reverting that last rollback, i.e. upgrading again to v0.35.1

### CODE REVIEW

**Comment by altergui** [comment]
@p4u what was the idea behind this?

https://github.com/vocdoni/vocdoni-node/blob/1a05e44910102ee96739ccc74fedb5e18dd6759c/vochain/app.go#L185-L197

i believe this was introduced by mistake during that merge commit 1a05e449, it doesn't even compile since `r, err := return app.Node.BroadcastTxSync()` but everything was commented out anyway, but i wonder why did you were touching that code, just in case i'm missing something.

i simply reverted that to 
https://github.com/vocdoni/vocdoni-node/blob/1cb556f4599aef0f451b580386fd2795d23d7408/vochain/app.go#L179-L181

**Comment by p4u** [comment]
> @p4u what was the idea behind this?
> 
> https://github.com/vocdoni/vocdoni-node/blob/1a05e44910102ee96739ccc74fedb5e18dd6759c/vochain/app.go#L185-L197
> 
> i believe this was introduced by mistake during that merge commit [1a05e44](https://github.com/vocdoni/vocdoni-node/commit/1a05e44910102ee96739ccc74fedb5e18dd6759c), it doesn't even compile since `r, err := return app.Node.BroadcastTxSync()` but everything was commented out anyway, but i wonder why did you were touching that code, just in case i'm missing something.
> 
> i simply reverted that to
> 
> https://github.com/vocdoni/vocdoni-node/blob/1cb556f4599aef0f451b580386fd2795d23d7408/vochain/app.go#L179-L181

I don't remember :/
But I'd say the txHash of the transaction is required for further operations. So probably this is why I did it this way.

**Comment by p4u** [comment]
BTW why you didn't bump to the last Tendermint version? I rolled it back because there were some bugs on the mempool layer, so I hope this is fixed now in new versions.

**Comment by p4u** [comment]
@altergui there are some conflicts that need to be fixed

**Comment by altergui** [comment]
> BTW why you didn't bump to the last Tendermint version? I rolled it back because there were some bugs on the mempool layer, so I hope this is fixed now in new versions.

i wanted first to undo the downgrade, to reproduce the original state and not make a mess of code changes.

i bumped to latest tendermint v0.35.9 and at least the ci passes. please test it, stress it, hope the mempool issues are fixed

**Comment by altergui** [comment]
> @altergui there are some conflicts that need to be fixed

(conflicts solved)

File: cmd/vochaininspector/inspector.go [hunk 1]
```diff
@@ -193,64 +187,65 @@ func newVochain(chain, dataDir string) *vochain.BaseApplication {
...
+	// 	addr, err := ethereum.AddrFromPublicKey(pubKey)
+	// 	if err != nil {
+	// 		log.Fatalf("cannot extract address from public key: %v", err)
+	// 	}
+	// 	vote.Nullifier = vochain.GenerateNullifier(addr, vote.ProcessId)
+	// 	// fmt.Printf("%+v\n", vote)
+	// 	vid, err := voteID(vote.ProcessId, vote.Nullifier)
+	// 	if err != nil {
+	// 		log.Fatalf("cannot get voteID: %v", err)
+	// 	}
```
Comment (mvdan) [inline cmd/vochaininspector/inspector.go]: delete the code instead? leaving it commented out doesn't help - it's always in the git history if we need it.

File: vochain/start.go [hunk 2]
```diff
@@ -376,251 +319,34 @@ func newTendermint(app *BaseApplication,
...
 	// If using a remote private validator, the pubkeys are only available after nm.NewNode()
 	// In that case, print them now, for debugging purposes
-	if len(localConfig.PrivValidatorListenAddr) > 0 {
-		pv := node.PrivValidator()
-		pubKey, err := pv.GetPubKey()
-		if err != nil {
-			log.Warnf("failed to get pubkey from private validator: %v", err)
-		}
-		log.Infof("tendermint address: %v", pubKey.Address())
+	/*	if len(localConfig.PrivValidatorListenAddr) > 0 {
```
Comment (mvdan) [inline vochain/start.go]: same

**Comment by p4u** [comment]
> > BTW why you didn't bump to the last Tendermint version? I rolled it back because there were some bugs on the mempool layer, so I hope this is fixed now in new versions.
> 
> i wanted first to undo the downgrade, to reproduce the original state and not make a mess of code changes.
> 
> i bumped to latest tendermint v0.35.9 and at least the ci passes. please test it, stress it, hope the mempool issues are fixed

Great.
But it would be nice if you can test it and stress it yourself. Since I'm currently focused on the APIv2.

File: cmd/vochaininspector/inspector.go (hunk 1 above)
Comment (altergui) [reply cmd/vochaininspector/inspector.go]: good catch: actually a no-op cli flag was being left behind, cleaned everything up in 1fadf6c

**Comment by p4u** [comment]
I'll review and merge this PR once we release Azeno next week. Thanks for the work @altergui

**Pull request merged**